packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/pools/components/pool-form/controller.js

```diff
@@ -2,16 +2,32 @@ import {
   DEFAULT_ALGORITHM,
   DEFAULT_SESSION_PERSISTENCE_TYPE,
   APP_COOKIE_SESSION_PERSISTENCE,
+  PROTOCOL_SESSION_PERSISTENCE_TYPE_COMBINATION,
 } from './constants';
 
 export default class OctaviaLoadBalancerPoolFormCtrl {
   $onInit() {
     this.APP_COOKIE_SESSION_PERSISTENCE = APP_COOKIE_SESSION_PERSISTENCE;
-    this.model.algorithm =
-      this.model.algorithm ||
-      this.algorithms.find(
-        (algorithm) => algorithm.value === DEFAULT_ALGORITHM,
+    this.model.algorithm = this.algorithms.find((algorithm) =>
+      this.model.algorithm
+        ? algorithm.value ===
+          (this.model.algorithm.value || this.model.algorithm)
+        : algorithm.value === DEFAULT_ALGORITHM,
+    );
```

Comment:
find method is hard to read
```suggestion
const algorithmExpectedValue = this.model.algorithm?.value || this.model.algorithm ||  DEFAULT_ALGORITHM;
    this.model.algorithm = this.algorithms.find(({value}) =>
     algorithmExpectedValue === value,
    );
```
Question : if `this.model.algorithm` is already set, is `.find()` call still necessary ?

Reply:
Sounds good to me 
```suggestion
    if (!this.model.algorithm.value) {
    this.model.algorithm = this.algorithms.find(({value}) =>
      value === (this.model.algorithm || DEFAULT_ALGORITHM),
    );
}
```